pkg/evm/evm.go

```diff
@@ -264,21 +264,31 @@ func FundAddress(
 	return nil
 }
 
-func GetSignedTxToMethodWithWarpMessage(
+func GetTxToMethodWithWarpMessage(
 	client ethclient.Client,
+	generateRawTxOnly bool,
+	from common.Address,
 	privateKeyStr string,
 	warpMessage *avalancheWarp.Message,
 	contract common.Address,
 	callData []byte,
 	value *big.Int,
 ) (*types.Transaction, error) {
 	const defaultGasLimit = 2_000_000
-	privateKey, err := crypto.HexToECDSA(privateKeyStr)
-	if err != nil {
-		return nil, err
+	var (
+		privateKey *ecdsa.PrivateKey
+		err        error
+	)
+	if privateKeyStr != "" {
+		privateKey, err = crypto.HexToECDSA(privateKeyStr)
+		if err != nil {
+			return nil, err
+		}
 	}
-	address := crypto.PubkeyToAddress(privateKey.PublicKey)
-	gasFeeCap, gasTipCap, nonce, err := CalculateTxParams(client, address.Hex())
+	if from == (common.Address{}) {
+		from = crypto.PubkeyToAddress(privateKey.PublicKey)
+	}
+	gasFeeCap, gasTipCap, nonce, err := CalculateTxParams(client, from.Hex())
```

Comment:
I don't foresee `from` being an empty address, especially if we are getting "validator manager owner" address on chain anyways. This means that we don't need `from` in argument and we can just use crypto.PubkeyToAddress(privateKey.PublicKey)

Reply:
this is for a user that just nows the address but not the private key. eg a multisig.

Reply:
to be used in general in cases where a user does not now a private key but want to create anyway an unsigned
tx. this evm library is more general than validator manager owners